src/common/base/abstractReportOnPromote.ts

```diff
@@ -49,33 +49,27 @@ export abstract class ReportOnPromoteCommand<T extends typeof SfCommand> extends
       : (this.flags['job-id'] as string);
     const org = this.flags['devops-center-username'] as Org;
 
-    try {
-      // query the deployment result related to the promote operation.
-      const operationResult: DeploymentResult = await selectOneDeploymentResultByAsyncJobId(
-        org.getConnection(),
-        asyncJobId
+    // query the deployment result related to the promote operation.
+    const operationResult: DeploymentResult | null = await selectOneDeploymentResultByAsyncJobId(
+      org.getConnection(),
+      asyncJobId
+    );
```

Comment:
```suggestion
    );
    if (!operationResult) {
      throw...
    }
```

Minor, just be very explicit about it.  I think it will be easier to follow the logic below.